editor/src/utils/utils.ts

```diff
@@ -933,6 +933,20 @@ function deduplicateBy<T>(key: (t: T) => string, ts: Array<T>): Array<T> {
   return results
 }
 
+export async function waitUntil(limitMS: number, check: () => boolean): Promise<boolean> {
+  if (check()) {
+    return true
+  } else if (limitMS <= 0) {
+    return false
+  } else {
+    return new Promise((resolve) => {
+      setTimeout(resolve, 5)
+    }).then(() => {
+      return waitUntil(limitMS - 5, check)
+    })
+  }
+}
+
```

Comment:
Please can you move this function to the `/src/core/shared/` directory rather than this file (which we really need to bin off)

Reply:
Fixed in 68683d84b.